client/quantum_serverless/core/provider.py

```diff
@@ -332,16 +335,127 @@ def get_compute_resources(self) -> List[ComputeResource]:
         return resources
 
     def create_compute_resource(self, resource) -> int:
-        """Create compute resource for provider."""
-        raise NotImplementedError
+        """
+        Create compute resource for provider.
+
+        First, create a compute_template based on the defined ComputeResource.
+        If the compute_template already exists (which shouldn't be the case,
+        exit silently. Otherwise, create the template. This template is
+        ephemeral and will be deleted when no longer needed (either the cluster
+        was created or we failed to create it and thus need to start over).
+
+        Then use that template to create a KubeRay Cluster. If the cluster
+        already exists, we exit silently. Users are only able to configure the
+        amount of cpu/memory and the number of worker replicas. The full spec
+        can be found in the KubeRay API spec under the "protoCluster"
+        definition:
+        https://github.com/ray-project/kuberay/blob/master/proto/swagger/cluster.swagger.json
+        """
+        compute_resource_name = self.name + "-template"
+
+        # Create template from resource
+        req = requests.get(
+            f"{self.host}/apis/v1alpha2/namespaces/{self.namespace}"
+            f"/compute_templates/{compute_resource_name}",
+            timeout=TIMEOUT,
+        )
+        if req.status_code == 200:
```

Comment:
```suggestion
        if req.ok:
```

I think this will cover correct redirects too